pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/impl/AggregateMetricsRule.java

```diff
@@ -68,6 +78,14 @@ private boolean shouldAggregate(InputManager inputManager) {
     return true;
   }
 
+  private Set<String> getGroupByColumnNames(QueryContext queryContext) {
+    List<ExpressionContext> groupByExprs = queryContext.getGroupByExpressions();
```

Comment:
This won't work if the group by expression is a transform function?

Reply:
Valid point. It's now updated.